app/src/androidTest/java/com/github/sdp/ratemyepfl/fragment/navigation/EventFragmentTest.kt

```diff
@@ -71,4 +82,33 @@ class EventFragmentTest {
         onView(withId(R.id.eventRecyclerView))
             .check(matches(isDisplayed()))
     }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun clickOnAddEventButtonLoggedInWorks() {
+        FakeConnectedUser.instance = FakeConnectedUser.Instance.FAKE_USER_1
+        HiltUtils.launchFragmentInHiltContainer<EventFragment> {}
+
+        init()
+        onView(withId(R.id.addEventButton)).perform(click())
+        intended(toPackage("com.github.sdp.ratemyepfl"))
+        release()
+    }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun clickOnAddEventButtonLoggedOutDoesNotWork() {
+        FakeConnectedUser.instance = FakeConnectedUser.Instance.LOGGED_OUT
+        val intent = Intent(ApplicationProvider.getApplicationContext(), MainActivity::class.java)
+        val scenario: ActivityScenario<MainActivity> = ActivityScenario.launch(intent)
+        onView(withId(R.id.activityMainBottomNavigationView)).perform(
+            navigateTo(R.id.event)
+        )
+
+        onView(withId(R.id.addEventButton)).perform(click())
+        Thread.sleep(2000)

```

Comment:
Is this sleep useful for the test ?

Reply:
Yes! It waits for the snackbar to disapear so the check on the add button is sure to pass.